app/javascript/controllers/signup_controller.js

```diff
@@ -46,340 +53,387 @@ passwordTooltip() {
       <li id="wh_space" class="invalid"><i class="fa fa-times mr-3 wh_space"></i>Not include blank spaces</li>
       <li id="nm_uid" class="invalid"><i class="fa fa-times mr-3 nm_uid"></i>Cannot contain username</li>
      </ul>
-  `
-  this.hideTooltips();
-  $('#user_password').attr('data-title', passwordHelper);
-  $('#user_password').tooltip('show');
-}
+  `;
+    this.hideTooltips();
+    $('#user_password').attr('data-title', passwordHelper);
+    $('#user_password').tooltip('show');
+  }
 
-hideTooltips() {
-  $('#user_oim_id').tooltip('hide');
-  $('#user_password').tooltip('hide');
-}
+  hideTooltips() {
+    $('#user_oim_id').tooltip('hide');
+    $('#user_password').tooltip('hide');
+  }
```

Comment:
I noticed almost everything in this controller was changed to vanilla js --> is it more optimal to use jQuery for tooltip-related UI? Just curious --> I've never seen any handling for tooltips with vanilla js

Reply:
It's ok to use either but we ultimately want to move away from jquery because it's just an extra package that we have to load. But, realistically, we won't be able to do that unless we plan it out and focus with rules.

Currently we don't have any rules for or against jquery so ¯\_(ツ)_/¯ 

In this particular file, the only code that was changed to fix the bug I mentioned above. All the other changes are from the prettier formatter.

Reply:
Brad is 100% right on this. I will add, we're moving away from jquery when writing things from scratch, so if we do get a project asking us to remove jquery it is easier. but it isn't a huge priority, just a nice to have